openquake/hazardlib/gsim/kuehn_2020.py

```diff
@@ -0,0 +1,761 @@
+        # epsilon for epistemic uncertainty
+        self.sigma_mu_epsilon = sigma_mu_epsilon
+        if self.sigma_mu_epsilon:
+            self.sigma_mu_model = _retrieve_sigma_mu_data(
+                self.DEFINED_FOR_TECTONIC_REGION_TYPE, self.region)
+        else:
+            self.sigma_mu_model = {}
+
+    def compute(self, ctx, imts, mean, sig, tau, phi):
+        """
+        See :meth:`superclass method
+        <.base.GroundShakingIntensityModel.compute>`
+        for spec of input and result values.
+        """
+        trt = self.DEFINED_FOR_TECTONIC_REGION_TYPE
+        if self.m_b:
+            # Take the user define magnitude scaling breakpoint
+            m_b = self.m_b
+        else:
+            # Take the global m_b for the tectonic region type and region
+            m_b = REGION_TERMS_IF[self.region]["mb"] \
+                if trt == const.TRT.SUBDUCTION_INTERFACE else \
+                REGION_TERMS_SLAB[self.region]["mb"]
+        C_PGA = self.COEFFS[PGA()]
+
+        # Get PGA on rock
+        pga1100 = np.exp(get_mean_values(C_PGA, self.region, trt, m_b,
+                                         ctx, ctx, ctx, None))
+        # For PGA and SA ( T <= 0.1 ) we need to define PGA on soil to
+        # ensure that SA ( T ) does not fall below PGA on soil
+        pga_soil = None
+        for imt in imts:
+            if ("PGA" in imt.string) or (("SA" in imt.string) and
+                                         (imt.period <= 0.1)):
+                pga_soil = get_mean_values(C_PGA, self.region, trt, m_b,
+                                           ctx, ctx, ctx, pga1100)
+                break
+
+        for m, imt in enumerate(imts):
+            # Get coefficinets for imt
+            C = self.COEFFS[imt]
+            m_break = m_b + C["dm_b"] if trt == const.TRT.SUBDUCTION_INTERFACE\
+                else m_b
+
+            if imt.string == "PGA":
+                mean[m] = pga_soil
+            elif ("SA" in imt.string) and (imt.period <= 0.1):
+                # If Sa (T) < PGA for T <= 0.1 then set mean Sa(T) to mean PGA
+                mean[m] = get_mean_values(C, self.region, trt, m_break,
+                                          ctx, ctx, ctx, pga1100)
+                idx = mean[m] < pga_soil
+                mean[m][idx] = pga_soil[idx]
+            else:
+                # For PGV and Sa (T > 0.1 s)
+                mean[m] = get_mean_values(C, self.region, trt, m_break,
+                                          ctx, ctx, ctx, pga1100)
+            # Apply the sigma mu adjustment if necessary
+            if self.sigma_mu_epsilon:
+                sigma_mu_adjust = get_sigma_mu_adjustment(self.sigma_mu_model,
+                                                          imt, ctx, ctx)
+                mean[m] += (self.sigma_mu_epsilon * sigma_mu_adjust)
+            # Get standard deviations
+            tau[m] = C["tau"]
+            phi[m] = C["phi"]
+        sig[:] = np.sqrt(tau ** 2.0 + phi ** 2.0)
+
```

Comment:
Since C["tau"] and C["phi"] are scalar it is better to write `sig[m] = np.sqrt(C['tau'] ** 2.0 + C['phi'] ** 2.0)`, but this is marginal.

Reply:
Done. Thanks!